src/Common/LRUCache.h

```diff
@@ -326,13 +376,29 @@ class LRUCache
             }
 
             const auto & cell = it->second;
-
-            current_size -= cell.size;
-            current_weight_lost += cell.size;
-
-            cells.erase(it);
-            queue.pop_front();
-            --queue_size;
+            auto evict_status = evict_policy.canRelease(*cell.value);// in default, it is CAN_EVITCT
+            switch (evict_status)
+            {
+                case LRUCacheEvictStatus::CAN_EVITCT:
```

Comment:
If there are only two states - CAN_EVICT and SKIP_EVICT, lets just make it a bool variable, no enum.

Reply:
done